libs/wingc/src/type_check.rs

```diff
@@ -1846,6 +1846,65 @@ impl<'a> TypeChecker<'a> {
 					_ => {}
 				};
 			}
+			StmtKind::TryCatch {
+				try_statements,
+				catch_block,
+				finally_statements,
+			} => {
+				// Create a new environment for the try block
+				let try_env = SymbolEnv::new(
+					Some(env.get_ref()),
+					self.types.void(),
+					false,
+					false,
+					env.flight,
+					stmt.idx,
+				);
+				try_statements.set_env(try_env);
+				self.inner_scopes.push(try_statements);
+
+				// Create a new environment for the catch block
+				if let Some(catch_block) = catch_block {
+					let mut catch_env = SymbolEnv::new(
+						Some(env.get_ref()),
+						self.types.void(),
+						false,
+						false,
+						env.flight,
+						stmt.idx,
+					);
+
+					// Add the exception variable to the catch block
+					if let Some(exception_var) = &catch_block.exception_var {
+						match catch_env.define(
+							exception_var,
+							SymbolKind::make_variable(self.types.string(), false, env.flight),
```

Comment:
Yes because of this: 
https://github.com/winglang/wing/blob/276611e0868b805e8bc771243b97ed4d61bd2cc1/libs/wingc/src/lib.rs#L180
and this:
https://github.com/winglang/wing/blob/276611e0868b805e8bc771243b97ed4d61bd2cc1/libs/wingc/src/jsify.rs#L604

Reply:
👍  Clean